ansible_ai_connect/ai/api/model_client/tests/test_wca_client.py

```diff
@@ -360,6 +360,26 @@ def test_playbook_gen_error(self):
             model_client.generate_playbook(request, text="Install Wordpress", create_outline=True)
             self.assertInLog("Caught retryable error after 1 tries.", log)
 
+    def test_playbook_gen_model_id(self):
+        self.assertion_count = 0
+        request = Mock()
+        model_client = WCAClient(inference_url="http://example.com/")
+        model_client.get_api_key = Mock(return_value="some-key")
+        model_client.get_token = Mock(return_value={"access_token": "a-token"})
+        model_client.session = Mock()
+
+        def get_my_model_id(user, organization_id, model_id):
+            self.assertEqual(model_id, "mymodel")
+            self.assertion_count += 1
+            return model_id
+
+        model_client.get_model_id = get_my_model_id
+
+        model_client.generate_playbook(
+            request, text="Install Wordpress", create_outline=True, model_id="mymodel"
+        )
+        self.assertGreater(self.assertion_count, 0)
```

Comment:
This is really only asserting that your mock `get_my_model_id` was called.

Would it not be better to check `WCAClient.get_model_id` was called?

Reply:
I wanted to assert `model_id` was passed to `get_model_id` when WCA playbook gen is used and thought if the mocked `get_model_id` received the given `model_id` value, it was asserted even though it is not the real `WCAClient.get_model_id`.

> Would it not be better to check WCAClient.get_model_id was called?

I thought I might add a DEBUG trace to `WCAClient.get_model_id` and check the log later, but did not take that option.  If it looks more adequate or there is a better different approach, I am fully open for the change.